viz-lib/src/visualizations/choropleth/hooks/useLoadGeoJson.js

```diff
@@ -0,0 +1,39 @@
+import { isString, isObject, get } from "lodash";
+import { useState, useEffect } from "react";
+import axios from "axios";
+import { visualizationsSettings } from "@/visualizations/visualizationsSettings";
+import createReferenceCountingCache from "@/lib/referenceCountingCache";
+
+const cache = createReferenceCountingCache();
+
+export default function useLoadGeoJson(mapType) {
+  const [geoJson, setGeoJson] = useState(null);
+  const [isLoading, setIsLoading] = useState(false);
+
+  useEffect(() => {
+    const mapUrl = get(visualizationsSettings, `choroplethAvailableMaps.${mapType}.url`, undefined);
+
+    if (isString(mapUrl)) {
+      setIsLoading(true);
+      let cancelled = false;
+
+      const promise = cache.get(mapUrl, () => axios.get(mapUrl).catch(() => null));
```

Comment:
This cache is to help with multiple references not reloading the json, right? (e.g.: in a dashboard)

Very interesting! 😁

Reply:
Yes, exactly. GeoJSON files are pretty large (300-500kB), and previously if there were a lot of visualizations - each of them was loading own copy of the same file. Also, this cache works on a Query View/Source pages - if there are multiple Choropleth visualizations with the same map.